components/datetime/src/format/datetime.rs

```diff
@@ -286,38 +324,64 @@ where
                     }
                 }
             };
-            format_number(w, value, field.length)?
+            format_number(
+                w,
+                fixed_decimal_format,
+                FixedDecimal::from(value),
+                field.length,
+            )?
         }
         FieldSymbol::Minute => format_number(
             w,
-            usize::from(
+            fixed_decimal_format,
+            FixedDecimal::from(usize::from(
                 datetime
                     .datetime()
                     .minute()
                     .ok_or(Error::MissingInputField)?,
-            ) as isize,
+            ) as isize),
             field.length,
         )?,
-        FieldSymbol::Second(Second::Second) => format_number(
-            w,
-            usize::from(
+        FieldSymbol::Second(Second::Second) => {
+            let mut seconds = FixedDecimal::from(usize::from(
                 datetime
                     .datetime()
                     .second()
                     .ok_or(Error::MissingInputField)?,
-            ) as isize,
-            field.length,
-        )?,
-        FieldSymbol::Second(Second::FractionalSecond) => format_number(
-            w,
-            usize::from(
-                datetime
-                    .datetime()
-                    .nanosecond()
-                    .ok_or(Error::MissingInputField)?,
-            ) as isize,
-            field.length,
-        )?,
+            ));
+            if let Some(PatternItem::Field(next_field)) = next_item {
+                if let FieldSymbol::Second(Second::FractionalSecond) = next_field.symbol {
+                    let mut fraction = FixedDecimal::from(usize::from(
+                        datetime
+                            .datetime()
+                            .nanosecond()
+                            .ok_or(Error::MissingInputField)?,
+                    ));
+
+                    // We only support fixed field length for fractional seconds.
+                    let precision = match next_field.length {
+                        FieldLength::Fixed(p) => p,
+                        _ => {
+                            return Err(Error::FixedDecimal);
```

Comment:
Suggestion: This is not really a "fixed decimal error"; it is an invalid pattern field. Consider using a different error type.